cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriter.java

```diff
@@ -99,6 +101,9 @@ public DatasetSystemMetadataWriter(MetadataStore metadataStore,
     if (description != null) {
       properties.put(DESCRIPTION, description);
     }
+    if (dataset instanceof FileSet || dataset instanceof PartitionedFileSet) {
```

Comment:
curious - where is the 'explore.enabled' tag placed for other types of datasets?